devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusBuild.java

```diff
@@ -78,6 +78,29 @@ public FileCollection getClasspath() {
                 .plus(mainSourceSet.getResources());
     }
 
+    @Input
+    public Map<String, String> getQuarkusBuildSytemProperties() {
+        Map<String, String> quarkusSystemProperties = new HashMap<>();
+        for (Map.Entry<Object, Object> systemProperty : System.getProperties().entrySet()) {
+            if (systemProperty.getKey() != null && systemProperty.getValue() instanceof String
```

Comment:
I think I would test the type of the key too. You never know what people do.

Reply:
yes you right. I added a test.